Src/Plugins/Tests.Security/Validators/SEC101_044.NpmCredentialsValidatorTests.cs

```diff
@@ -35,43 +35,60 @@ public void NpmCredentialsValidator_MockHttp()
 
             string fingerprintText = $"[host={host}][id={id}][secret={secret}]";
 
+            var authorizedResponse = new HttpResponseMessage(HttpStatusCode.OK);
+            var notFoundResponse = new HttpResponseMessage(HttpStatusCode.NotFound);
+            var unauthorizedResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized);
```

Comment:
>Unauthorized [](http://example.com/codeflow?start=78&length=12)

Where is your case for NonAuthoritativeInformation?

Reply:
I can add it, but, its the same codepath as OK/NotFound/NonAuhotritativeInformation.

Reply:
? It is not the same code path, it's a separate conditional. Am I looking at the wrong code?

                if (responseWithNoCredentials.StatusCode == HttpStatusCode.OK ||
                    responseWithNoCredentials.StatusCode == HttpStatusCode.NotFound ||
                    responseWithNoCredentials.StatusCode == HttpStatusCode.NonAuthoritativeInformation)
                {
                    return ValidationState.NoMatch;
                }

Reply:
just added and updated the title to be clear of which is which.

Reply:
I don't understand what you're saying. Was it useful to add this test or not?

Reply:
we have  two occasions where the API return OK:
1. API does not need any credential
2. API does need credential and we used the correct

The case (1) is also the same code path when the API does not exist (NotFound) or some redirect (like AzureDevOps NonAuthoritativeInformation). With that, my first test was already covering this where i was looking for OK when no credential was required.

Adding the other two options (notfound and NonAuthoritativeInformation) adds the branch coverage, but the code path was already coverage.

That's the only difference. :)

Reply:
Branch coverage *is* code coverage. :) I see your point, the literal blocks within the conditional expression had been covered, but the testing was incomplete because you hadn't comprehensively verified that each relevant condition resulted in executing those blocks. Search the internet for 'condition decision coverage'. This is the typical measure of code coverage for most test tools.